src/components/data/details/PermissionsPanel.js

```diff
@@ -74,32 +90,67 @@ function PermissionsTabPanel(props) {
         return merged;
     };
 
-    // The permissions endpoint returns only the user ID and the permission value.
-    // Fetch the rest of each user's details.
-    // Without useCallback, useEffect thinks this function is constantly
-    // updating and causes infinite re-renders
-    const fetchUserInfo = useCallback((permissions) => {
-        const users = new Set(permissions.map((item) => item.user));
-        getUserInfo([...users]).then((userInfos) => {
-            const mergedInfo = mergeUsersWithPerms(permissions, userInfos);
-            setPermissions(mergedInfo);
-            setLoading(false);
-        });
-    }, []);
+    const [fetchUserInfo, { status: fetchUserInfoStatus }] = useMutation(
+        getUserInfo,
+        {
+            onSuccess: (userInfos, { permissions }) => {
+                const mergedInfo = mergeUsersWithPerms(permissions, userInfos);
+                setPermissions(mergedInfo);
+            },
+            onError: () =>
+                announce({
+                    text: formatMessage(intl, "fetchPermissionsError"),
+                    variant: "error",
+                }),
+        }
+    );
```

Comment:
Yeah, this might be my own lack of understanding, but it seems like `useMutation` doesn't really have a functional difference between `useQuery`.  The docs say it's typically used when you need data to be modified, but I found it to be really useful in controlling when a request gets executed instead.  I did see the alternative was to set the key to a falsy value. I personally found using `useMutation` to be more straightforward, but I can switch to the falsy key method instead.

Reply:
Agreed. `useQuery` and `useMutation` can both be used here. I just want to avoid any sort of confusion in future when we look into the code. Also some consistency in our approach.